### PR TITLE
teleconsole 0.4.0

### PR DESCRIPTION
**Pull request by chenrui333**

### CODE REVIEW

**Comment by fxcoudert** [comment]
- Patch needs to be removed
- Go resources are deprecated

File: Formula/teleconsole.rb [hunk 1]
```diff
@@ -29,7 +30,7 @@ class Teleconsole < Formula
 
   go_resource "github.com/gravitational/teleport" do
     url "https://github.com/gravitational/teleport.git",
-        :revision => "002b640a16f097e2f834b4ae33c9edfb81d5798c"
+        :tag => "v2.0.0-alpha.4"
```
Comment (chenrui333) [inline Formula/teleconsole.rb]: relates to these lines, https://github.com/gravitational/teleconsole/blob/master/Makefile#L36-L38

**Comment by chenrui333** [comment]
Fixed the issue.

File: Formula/teleconsole.rb (hunk 1 above)
Comment (zbeekman) [reply Formula/teleconsole.rb]: we should not be shipping an alpha version

**Comment by zbeekman** [comment]
> Fixed the issue.

We need to be using go 1.11 modules instead of resources, and we cannot ship an alpha version.

File: Formula/teleconsole.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/teleconsole.rb]: Yeah, I think that is the stuff in their codebase, maybe I should PR to fix there?

Comment (zbeekman) [reply Formula/teleconsole.rb]: @chenrui333 yess please, and also we should be using go modules.

Comment (chenrui333) [reply Formula/teleconsole.rb]: Please note, their project does not support `go modules` yet, i.e. the `go.mod` and `go.sum` are not there.

So I think we have to wait until they do the support.

File: Formula/teleconsole.rb [hunk 2]
```diff
@@ -3,8 +3,9 @@
 class Teleconsole < Formula
   desc "Free service to share your terminal session with people you trust"
   homepage "https://www.teleconsole.com"
-  url "https://github.com/gravitational/teleconsole/archive/0.3.1.tar.gz"
-  sha256 "663307a1dfe4baadf7e1ed9f5b66b1d203bf9696068e9bcd86e535f286e64d59"
+  url "https://github.com/gravitational/teleconsole/archive/0.4.0.tar.gz"
+  sha256 "ba0a231c5501995e2b948c387360eb84e3a44fe2af6540b6439fc58637b0efa4"
+  head "https://github.com/gravitational/teleconsole.git"
```
Comment (zbeekman) [inline Formula/teleconsole.rb]:
```suggestion
```

Please remove HEAD!

**Review by zbeekman — CHANGES_REQUESTED** [review]

File: Formula/teleconsole.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/teleconsole.rb]: 👍

**Comment by zbeekman** [comment]
@fxcoudert what are your thoughts concerning go resources? I've very little go experience and I don't see an obvious way to achieve this from the upstream repo.

**Comment by fxcoudert** [comment]
Because it's an update, and I don't know how to transition it: ok for now

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by chenrui333** [comment]
yeah, I think we will need to wait for the upstream repo to update the build tool to `go modules` (maybe), so that we can deprecate the go resources usage.

**Pull request closed without merging**